org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/codegen/CodeStream.java

```diff
@@ -7885,11 +7873,30 @@ public void addPatternAccessorExceptionHandler(BlockScope scope, boolean addTarg
 	if (target != null) {
 		target.place();
 	}
-	if (scope instanceof MethodScope && localsToReinit.size() > 0) {
-		for (LocalVariableBinding local : localsToReinit) {
-			int pos = this.position;
-			local.recordInitializationStartPC(pos);
+	reinitLocals(scope, localsToReinit);
+}
+public void reinitLocals(BlockScope scope, List<LocalVariableBinding> localsToReinit) {
```

Comment:
I think *Scope is a better place for these methods (reinitLocals and getLocalsToReinit).